File: Problems/0405_Convert_a_Number_to_Hexadecimal/Project_CS/Convert_a_Number_to_Hexadecimal.cs
using System;
using System.Collections.Generic;

public class Solution {
    public string ToHex(int num) {
        uint temp;

        if (num > 0)
            temp = (uint)num;
        else if (num == 0)
            return "0";
        else
            temp = (uint)Math.Pow(2,32) + (uint)num;
        //  temp = (uint)(4294967296 + num);
        
        string chars = "0123456789abcdef";
        uint modded;
        string resultStr = "";

        while (temp > 0) {
            modded = temp % 16;
            resultStr = chars[(int)modded] + resultStr;
            temp /= 16;
        }

        return resultStr;
    }

    public void Main(string args)
    {
        int num = int.Parse(args);

        Console.WriteLine("num = " + num.ToString());

        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();
        
        string result = ToHex(num);

        Console.WriteLine("result = " + result);
        
        sw.Stop();
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms\n");
    }
}

File: Problems/0400_0499/0405_Convert_a_Number_to_Hexadecimal/Project_Python3/Convert_a_Number_to_Hexadecimal.py
# coding: utf-8

import os
import sys
import time

class Solution:
    def toHex(self, num):
        """
        :type num: int
        :rtype: str
        """
        chars = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d", "e", "f"]
        resultStr = ""

        if num > 0:
            temp = num
        elif num == 0:
            return "0"
        else:
            temp = 2**32 + num

        while temp > 0:
            modded = temp % 16
            resultStr = chars[modded] + resultStr
            temp //= 16
        return resultStr

def main():
    argv = sys.argv
    argc = len(argv)

    if argc < 2:
        print("Usage: python {0} <testdata.txt>".format(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("{0} not found...".format(argv[1]))
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        temp = temp.strip()
        if temp == "":
            continue
        print("args = {0}".format(temp))
        loop_main(temp)
    #   print("Hit Return to continue...")
    #   input()

def loop_main(temp):
    num = int(temp)

    sl = Solution()
    time0 = time.time()
    result = sl.toHex(num)
    print("result = {0}".format(result))

    time1 = time.time()

    print("Execute time ... : {0:f}[s]\n".format(time1 - time0))

if __name__ == "__main__":
    main()
